src/dune_rules/mdx.ml

```diff
@@ -229,17 +271,98 @@ let gen_rules_for_single_file stanza ~sctx ~dir ~expander ~mdx_prog src =
   Super_context.add_alias_action sctx (Alias.runtest ~dir) ~loc:(Some loc) ~dir
     diff_action
 
+let name = "mdx_gen"
+
+let mdx_prog_gen t ~sctx ~dir ~scope ~expander ~mdx_prog =
+  let loc = t.loc in
+  let dune_version = Scope.project scope |> Dune_project.dune_version in
+  let file = Path.Build.relative dir "mdx_gen.ml-gen" in
+
+  (* Libs from the libraries field should have their include directories sent to
+     mdx *)
+  let open Resolve.O in
+  let directory_args =
+    let+ libs_to_include =
+      Resolve.List.filter_map t.libraries ~f:(function
+        | Direct lib
+        | Re_export lib ->
+          let+ lib = Lib.DB.resolve (Scope.libs scope) lib in
+          Some lib
+        | _ -> Resolve.return None)
+    in
+
+    let mode = Context.best_mode (Super_context.context sctx) in
+
+    let libs_include_paths = Lib.L.include_paths libs_to_include mode in
+
+    let open Command.Args in
+    let args =
+      Path.Set.to_list libs_include_paths
+      |> List.map ~f:(fun p -> S [ A "--directory"; Path p ])
+    in
+    S args
+  in
+
+  let prelude_args =
+    Command.Args.S (List.concat_map t.preludes ~f:(Prelude.to_args ~dir))
+  in
+
+  (* We call mdx to generate the testing executable source *)
+  let action =
+    Command.run ~dir:(Path.build dir) mdx_prog ~stdout_to:file
+      [ A "dune-gen"; prelude_args; Resolve.args directory_args ]
+  in
+  let open Memo.Build.O in
+  let* () = Super_context.add_rule sctx ~loc ~dir action in
+
+  (* We build the generated executable linking in the libs from the libraries
+     field *)
+  let obj_dir = Obj_dir.make_exe ~dir ~name in
+  let main_module_name = Module_name.of_string name in
+  let module_ = Module.generated ~src_dir:(Path.build dir) main_module_name in
+  let modules = Modules.singleton_exe module_ in
+  let flags = Ocaml_flags.default ~dune_version ~profile:Release in
+  let compile_info =
+    Lib.DB.resolve_user_written_deps_for_exes (Scope.libs scope)
+      [ (t.loc, name) ]
+      (Lib_dep.Direct (loc, Lib_name.of_string "mdx.test") :: t.libraries)
+      ~pps:[ (*Preprocess.Per_module.pps t.preprocess*) ] ~dune_version
```

Comment:
is that comment still relevant?

Reply:
It's not, there is no need to preprocess the source generated by mdx for the test executable.